lib/vast/Conversion/ABI/LowerABI.cpp

```diff
@@ -217,19 +217,14 @@ namespace vast
                 auto as_lvalue = [&]() -> mlir::Operation *
                 {
                     VAST_ASSERT(val->getNumResults() == 1);
-                    auto as_val = val->getResult(0);
-                    if (mlir::isa< hl::LValueType >(as_val.getType()))
-                        return val;
-
-                    if (auto implicit_cast = mlir::dyn_cast< hl::ImplicitCastOp >(val))
-                    {
-                        if (implicit_cast.getKind() == hl::CastKind::LValueToRValue)
-                        {
-                            return implicit_cast.getOperand().getDefiningOp();
-                        }
-                    }
-
-                    VAST_UNREACHABLE("ABI conversion could not make lvalue for {0}", *val);
+                    auto mctx = direct.getContext();
+                    auto type = hl::PointerType::get(
+                        mctx,direct.getOperand(0).getType());
```

Comment:
spacing

Reply:
I will run clang-format on the whole `abi` component, but it will be done in a separate PR. Fixed this issue.